data/api/client-server/cross_signing.yaml

```diff
@@ -19,11 +19,26 @@ paths:
   /keys/device_signing/upload:
     post:
       x-addedInMatrixVersion: "1.1"
+      x-changedInMatrixVersion:
+        "1.11": UIA is not always required for this endpoint.
       summary: Upload cross-signing keys.
       description: |-
         Publishes cross-signing keys for the user.
 
         This API endpoint uses the [User-Interactive Authentication API](/client-server-api/#user-interactive-authentication-api).
+        
+        User-Interactive Authentication is required, except in these cases:
+        - there is no existing cross-signing master key uploaded to the homeserver, OR
+        - there is an existing cross-signing master key and it exactly matches the
+          cross-signing master key provided in the request body. If there are any additional
+          keys provided in the request (self-signing key, user-signing key) they MUST also
+          match the existing keys stored on the server. In other words, the request contains
+          no new keys.
```

Comment:
I copied this section from the MSC, but it seems to ignore the case where a client uploads the master key and the other keys (self-signing and user-signing) separately.

My understanding according to the paragraph below is that setting the other keys if they were not set before, or re-uploading the same keys, should not require UIA, even if the master key is not part of the request (i.e. it was already uploaded before).